odl/tomo/geometry/parallel.py

```diff
@@ -125,84 +171,222 @@ def det_to_src(self, angles, dpar, normalized=True):
             raise ValueError('`angles` {} not in the valid range {}'
                              ''.format(angles, self.motion_params))
-
         if dpar not in self.det_params:
             raise ValueError('`dpar` {} not in the valid range '
                              '{}'.format(dpar, self.det_params))
-
         if not normalized:
             raise NotImplementedError('non-normalized detector to source is '
                                       'not available in parallel case')
 
         return self.rotation_matrix(angles).dot(self.detector.normal)
 
 
-class Parallel2dGeometry(ParallelGeometry):
+class Parallel2dGeometry(ParallelBeamGeometry):
 
     """Parallel beam geometry in 2d.
 
     The motion parameter is the counter-clockwise rotation angle around
-    the origin, and the detector is a line detector perpendicular to the
-    ray direction.
+    the origin, and the detector is a line detector.
 
-    In the standard configuration, the detector reference point starts
-    at ``(1, 0)``, and the initial detector axis is ``(0, 1)``.
+    In the standard configuration, the detector is perpendicular to the
+    ray direction, its reference point is initially at ``(0, 1)``, and
+    the initial detector axis is ``(1, 0)``.
+
+    For details, check `the online docs
+    <https://odlgroup.github.io/odl/guide/geometry_guide.html>`_.
     """
 
-    def __init__(self, apart, dpart, **kwargs):
+    _default_config = dict(det_pos_init=(0, 1), det_axis_init=(1, 0))
+
+    def __init__(self, apart, dpart, det_pos_init=(0, 1), **kwargs):
         """Initialize a new instance.
 
         Parameters
         ----------
         apart : 1-dim. `RectPartition`
-            Partition of the angle interval
+            Partition of the angle interval.
         dpart : 1-dim. `RectPartition`
-            Partition of the detector parameter interval
-        det_init_pos : `array-like`, shape ``(2,)``, optional
-            Initial position of the detector reference point. The zero
-            vector is only allowed if ``det_init_axis`` is explicitly
-            given.
-            Default: ``(1, 0)``.
-        det_init_axis : `array-like` (shape ``(2,)``), optional
-            Initial axis defining the detector orientation.
-            By default, a normalized `perpendicular_vector` to
-            ``det_init_pos`` is used, which is only valid if
-            ``det_init_axis`` is not zero.
+            Partition of the detector parameter interval.
+        det_pos_init : `array-like`, shape ``(2,)``, optional
+            Initial position of the detector reference point.
+            Default: ``(0, 1)``.
+        det_axis_init : `array-like` (shape ``(2,)``), optional
+            Initial axis defining the detector orientation. The default
+            depends on ``det_pos_init``, see Notes.
+
+        Other Parameters
+        ----------------
+        init_matrix : `array_like`, shape ``(2, 2)``, optional
```

Comment:
Overall, the more I read this, I'm getting worried about this `init_matrix` and `translation` stuff clogging the interface while violating both DRY and "there should preferably only be one way to do it". I can certainly see why we want it, but I'm afraid it is clogging the interfaces for a use-case used by very few.

Would it somehow be possible to re-factor the geometries so that these are not needed, or perhaps create free functions that create objects equivalent to these? Another option would perhaps be to add some kind of mixing wraper that "adds" these to another geometry, e.g. `my_modified_geometry(parallel_beam_geometry(space), translation=[0, 1])`?

Another option would be to expand the library of "default geometries" like `parallel_beam_geometry` and somehow accept that these are intended to be the "general backend" and not something people will use by default.

I'll avoid reviewing these specific parts in depth until we have a word on this.